mlir-tensorrt/compiler/lib/Dialect/Plan/Transforms/CreateClosedRegions.cpp

```diff
@@ -503,11 +430,162 @@ static LogicalResult createClosedGroupOp(RewriterBase &rewriter,
       return failure();
     inputAttrs.push_back(boundsAttr);
   }
-  closedGroupOp.setInputAttrsAttr(inputAttrs);
 
+  return inputAttrs;
+}
+
+static LogicalResult materializeDestinationOperands(
+    RewriterBase &rewriter, plan::InlineGroupOp op, DataFlowSolver &solver,
+    SmallVector<DestinationOperandMaterializationResult> &destinationOperands) {
+  destinationOperands.reserve(op.getNumResults());
+  for (OpResult res : op->getOpResults()) {
+    FailureOr<DestinationOperandMaterializationResult> destResult =
+        materializeDestinationOperand(rewriter, res.getLoc(), op,
+                                      res.getResultNumber(), solver);
+    if (failed(destResult)) {
+      return emitError(res.getLoc())
+             << "failed to materialize destination operand of type "
+             << res.getType();
+    }
+    destinationOperands.push_back(*destResult);
+  }
   return success();
 }
 
+// Helper function to create DPS closed group op.
+static LogicalResult createInlineClosedGroupOp(
+    RewriterBase &rewriter, plan::InlineGroupOp op, DataFlowSolver &solver,
+    const SmallVector<Value> &inputs,
+    const SmallVector<DestinationOperandMaterializationResult>
+        &destinationOperands) {
+  InlineClosedGroupOp closedGroupOp = rewriter.create<InlineClosedGroupOp>(
+      op.getLoc(), /*target=*/op.getTarget(),
+      /*inputs=*/inputs,
+      /*outs=*/
+      llvm::map_to_vector(destinationOperands,
+                          [](const auto &x) { return x.destinationOperand; }));
+
+  rewriter.inlineBlockBefore(
+      &op.getRegion().front(), &closedGroupOp.getRegion().front(),
+      closedGroupOp.getRegion().front().end(),
+      closedGroupOp.getRegion().getArguments().take_front(
+          op.getRegion().getNumArguments()));
+
+  SmallVector<Value> replacements;
+  replacements.reserve(destinationOperands.size());
+  for (auto [newResult, destOperand, originalType] :
+       llvm::zip(closedGroupOp->getResults(), destinationOperands,
```

Comment:
Use `zip_equal` if all lists are supposed to be the same size (and it will assert as such). Otherwise the range is limited by the smallest element.